src/osd/ECCommon.cc

```diff
@@ -807,6 +816,12 @@ int ECCommon::ReadPipeline::send_all_remaining_reads(
   const hobject_t &hoid,
   ReadOp &rop)
 {
+  //FIXME: This function currently assumes that if it has already read a shard
+  //       then no further reads from that shard are required.  However with
+  //       the experimental optimised partial reads, it is possible for extra
+  //       reads to be required to an already-read shard. We plan on fixing this
+  //       before allowing such a configuration option to be enabled outside
+  //       test/dev environments.
```

Comment:
You could assert here if  cct->_conf->osd_ec_partial_reads_experimental is set to ensure the code stops cleanly when an unimplemented code path tries to run

Reply:
Added assert anyway.